### ISSUE TITLE
Workload status should indicate which component is failing, if known

### ISSUE THREAD

**Issue by idoru** [vmware-tanzu/cartographer#260]
### Description of problem

Let's say you have a component template in your supply chain which is somehow invalid, e.g. missing `kind`. Further to this, perhaps it is also missing a `name`. When a `workload` using such a supply chain tries to stamp the template, it enters this observed status:

```
Status:
  Conditions:
    Last Transition Time:  2021-10-21T17:20:50Z
    Message:
    Reason:                Ready
    Status:                True
    Type:                  SupplyChainReady
    Last Transition Time:  2021-10-21T17:20:58Z
    Message:               unable to apply object 'my-applications/': list: Object 'Kind' is missing in 'unstructured object has no kind'
    Reason:                TemplateRejectedByAPIServer
    Status:                False
    Type:                  ComponentsSubmitted
    Last Transition Time:  2021-10-21T17:20:58Z
    Message:               unable to apply object 'my-applications/': list: Object 'Kind' is missing in 'unstructured object has no kind'
    Reason:                TemplateRejectedByAPIServer
    Status:                False
    Type:                  Ready
  Observed Generation:     1
  Supply Chain Ref:
    Kind:  ClusterSupplyChain
    Name:  source-test-to-url
Events:    <none>
```

It seems clear that a template is missing a Kind but without a name there's no longer any unique context to identify which component is involved.

### Proposed solution

```gherkin
Given a supply chain with a template that is not valid
When a workload is reconciled for the supply chain
Then the observed status for the workload shows the name of the component in addition to the reason why it failed.
```

### Example
Create an otherwise valid supply chain, but substitute this for the source template:
```
---
apiVersion: carto.run/v1alpha1
kind: ClusterSourceTemplate
metadata:
  name: git-template---templates-refer-to-workload
spec:
  urlPath: .data.workload_git_url
  revisionPath: .data.workload_git_ref_branch
  template: {}
```

Apply a valid workload for the supply chain and observe the failure shown in the issue description.

**Comment by emmjohnson** [comment]
We think this is fixed but let's double check

**Comment by squeedee** [comment]
@idoru is going to review if this is resolved.

**Comment by idoru** [comment]
Looking at `main` right now and this condition does not indicate the name of the resource.

It will include the name of the stamped resource, but in the case where there may be mistakes in the templates, these names could be misleading, ambiguous or missing.

Granted, I think we hope these kinds of mistakes are few and far between, but when I wrote this story I thought it might be kind to users if we also included the resource name from the supply chain.
